lib/thrift/binary/framed/protocol_handler.ex

```diff
@@ -11,12 +11,13 @@ defmodule Thrift.Binary.Framed.ProtocolHandler do
   }
   require Logger
 
-  @spec start_link(reference, pid, module, {module, module}) :: GenServer.on_start
+  @spec start_link(reference, pid, module, {module, module, Keyword.t}) :: GenServer.on_start
```

Comment:
Is it worth defining a type for tcp_opts?  Can be helpful for documentation purposes.